### DISCUSSION TITLE
Using this package as a dependency?

### DISCUSSION THREAD

**Discussion by arogozhnikov** [microsoft/vscode-jupyter#10121] [Questions and Answers] [answered]
Hi, I'm writing a custom NotebookSerializer with typescript.

I want to be able to use some functions from this package (in particular, output conversion). 
Is there a way I can specify this package in dependencies? 

I've tried `npm install https://github.com/microsoft/vscode-jupyter` and it fails.

PS. I'm not deep into node, maybe I'm missing something obvious

**Comment by arogozhnikov** [comment]
In particular, that's the thing I can't overcome:

```
$ npm install https://github.com/microsoft/vscode-jupyter --save
npm WARN deprecated crypto@1.0.1: This package is no longer supported. It's now a built-in Node module. If you've depended on crypto, you should switch to the one that's built-in.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated vscode-debugprotocol@1.51.0: This package has been renamed to @vscode/debugprotocol, please update to the new name
npm WARN deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
npm ERR! code 127
npm ERR! path /Users/axelr/projects/vscode_extension/vscode-newpyter-js/node_modules/jupyter
npm ERR! command failed
npm ERR! command sh -c npm run download-api && node ./build/ci/postInstall.js
npm ERR! > jupyter@2022.5.100 download-api
npm ERR! > vscode-dts dev
npm ERR! sh: vscode-dts: command not found

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/axelr/.npm/_logs/2022-05-25T01_57_54_108Z-debug-0.log

```

**Comment by DonJayamanne** [comment, marked as answer]
Unfortunately the Jupyter extension isn't an npm package that you could install. Instead its wrtiten to be used as an extension within VS Code.

> Hi, I'm writing a custom NotebookSerializer with typescript.
> I want to be able to use some functions from this package (in particular, output conversion).
> Is there a way I can specify this package in dependencies?

For this, you'd have to copy the code.
